csharp/src/Apache.Arrow/Memory/NativeMemoryAllocator.cs

```diff
@@ -41,14 +42,10 @@ protected override Memory<byte> AllocateInternal(int length, out int bytesAlloca
 
             GC.AddMemoryPressure(bytesAllocated);
 
-            return manager.Memory;
-        }
+            // Ensure all allocated memory is zeroed.
```

Comment:
> why do you zero the memory? I would expect that it might have some random values, but they are getting overwritten anyway by the reader later.

I was responsible for this. It was useful during development to identify padding when inspecting buffers (since all padding is zeroed), and at the time we desired the same memory allocation semantics as general .NET primitives. I still see the value in zeroing the memory, but perhaps it can be placed behind a flag and defaulted to false for a performance advantage (if any)?

Reply:
I was just moving this zeroing from the base class (which would happen for ANY allocator) to just this native memory case. This way a managed memory implementation doesn't need to re-zero the memory, if it doesn't want to. Can you open a JIRA for this?

Reply:
Thanks for the explanation. If it shows up in the profiles in the next performance investigation we can consider removing it then. #resolved

Reply:
I've opened https://issues.apache.org/jira/browse/ARROW-5276 to add this option.